File: test/fixtures/jdt_test/src/main/java/com/example/Widget.java
package com.example;

/**
 * Simple fixture type for JdtBridge integration tests.
 */
public class Widget {
    private int state;

    public Widget() {
        this.state = 0;
    }

    public int compute(int x) {
        return x + 1;
    }

    public int addState(int delta) {
        this.state += delta;
        return this.state;
    }
}

File: test/fixtures/pyright_test/widget.py
"""Fixture module exercised by PyrightBridge integration tests."""


class Widget:
    """A simple widget with an integer state."""

    def __init__(self, x: int) -> None:
        self.x = x

    def compute(self) -> int:
        """Return the widget's value plus one."""
        return self.x + 1


def make_widget(x: int) -> Widget:
    """Factory for a :class:`Widget`."""
    return Widget(x)


def sum_widgets(count: int) -> int:
    """Sum the computed values of ``count`` widgets."""
    total = 0
    for i in range(count):
        total += make_widget(i).compute()
    return total
